lib/Onyx.js

```diff
@@ -794,6 +802,27 @@ function notifyCollectionSubscribersOnNextTick(key, value) {
     Promise.resolve().then(() => keysChanged(key, value));
 }
 
+/**
+ * This method waits for prior modifications to complete before running the next modification.
+ *
+ * @param {Function} runModification
+ * @param {String} key If "key" is not set (or set to ALL_KEYS_OPERATION) then the modification will be queued
+ * until all prior modifications complete. After that, a new modification with key -1 will be queued.
+ * @returns {Promise}
+ */
+function queueModification(runModification, key = ALL_KEYS_OPERATION) {
```

Comment:
Agree, this is a more obvious approach. The reason why i added this, is because we want to be able to

1. wait with `clear` to execute until all previous write operations are complete (because e.g. `merge` could overwrite `clear` if it takes longer)
2. wait with all write operations for `clear` to finish, because otherwise `clear` might remove certain keys afterwards again